### ISSUE TITLE
logpdf(Binomial(0,0), x=0) returns NaN instead of 0 when x is Int32 or lower

### ISSUE THREAD

**Issue by BenjamenSimon** [JuliaStats/Distributions.jl#1397]
```julia
logpdf(Binomial(0, 0.), 0) # = 0.00

logpdf(Binomial(0, 0.), 0.) # = 0.00

logpdf(Binomial(0, 0.), Int32(0)) # = NaN

logpdf(Binomial(0, 0.), Int16(0)) # = NaN

logpdf(Binomial(0, 0.), Float32(0)) # = NaN

logpdf(Binomial(0, 0.), Float16(0)) # = NaN
```

**Comment by andreasnoack** [comment]
I believe this will be fixed by https://github.com/JuliaStats/StatsFuns.jl/pull/113. `Binomial(0, 0.)` is a really odd distribution, though.

**Comment by devmotion** [comment]
This is also fixed by https://github.com/JuliaStats/StatsFuns.jl/pull/125 (BTW would be great if you can find time for a review @andreasnoack - otherwise, if you are busy maybe someone else could review it?):
```julia

julia> logpdf(Binomial(0, 0.), 0) # = 0.00
0.0

julia> logpdf(Binomial(0, 0.), 0.) # = 0.00
0.0

julia> logpdf(Binomial(0, 0.), Int32(0)) # = NaN
0.0

julia> logpdf(Binomial(0, 0.), Int16(0)) # = NaN
0.0

julia> logpdf(Binomial(0, 0.), Float32(0)) # = NaN
0.0

julia> logpdf(Binomial(0, 0.), Float16(0)) # = NaN
0.0
```

**Comment by BenjamenSimon** [comment]
Thank you both.. Do you have a timeline of when those will be implemented/fixed?

For context, its one term in the likelihood of a chain-binomial simulation. So it represents the probability that no events occur when there are no events that can occur (which is 1) which is a valid situation in the model of interest.

**Comment by devmotion** [comment]
As a temporary workaround you can install https://github.com/JuliaStats/StatsFuns.jl/pull/125 manually by running
```julia
] add StatsFuns#dw/rmath_real+julia_logpdf
```
in the Julia REPL. Unfortunately, the PR is missing a more detailed review (should work fine though) and it is difficult to estimate when it will be approved, merged, and released.

**Comment by devmotion** [comment]
The issue will be fixed in StatsFuns 0.9.11 which will be available when https://github.com/JuliaRegistries/General/pull/45672 is merged. Once it is available you should revert the manual installation suggested in the last comment (run `] free StatsFuns` in the Julia REPL) and install the latest version of StatsFuns (e.g. by running `] up`).
